utils.py

```diff
@@ -889,12 +889,30 @@ def check_auth(user, hutch):
     return False
 
 def check_special(ioc, hutch):
-    lines = open(SPECIAL_FILE % hutch).readlines()
-    lines = [l.strip() for l in lines]
-    for l in lines:
-        if l == ioc:
-            return True
-    return False
+    with open(SPECIAL_FILE % hutch) as fp:
+        lines = fp.readlines()
+        lines = [l.strip() for l in lines]
+        for l in lines:
+            if l == ioc:
+                return True
+        return False
+
+# checks if ioc is marked as toggleable between defined versions of IOCs
+# schema will be ioc_name:permittedversion1,permittedversion2,etc
+def check_special_toggle(ioc, hutch, version):
+    # check that this ioc for this hutch is permissioned specially
+    if not check_special(ioc, hutch):
```

Comment:
strip isn't going to take off the ":" and after.  So check_special is going to fail if you have allowable upgrades.

Reply:
fixing it in check special to parse around the `:`